src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java

```diff
@@ -993,39 +998,37 @@ public CompactionTasks getUserDefinedTasks(Collection<SSTableReader> sstables, i
         }
     }
 
+    @Override
     public int getEstimatedRemainingTasks()
     {
-        maybeReloadDiskBoundaries();
-        int tasks = 0;
-        readLock.lock();
-        try
-        {
-            for (AbstractCompactionStrategy strategy : getAllStrategies())
-                tasks += strategy.getEstimatedRemainingTasks();
-        }
-        finally
-        {
-            readLock.unlock();
-        }
-        return tasks;
+        return getStrategies(false).stream()
+                                   .flatMap(list -> list.stream())
+                                   .map(CompactionStrategy::getEstimatedRemainingTasks)
+                                   .reduce(0, Integer::sum);
     }
 
-    public boolean shouldBeEnabled()
+    @Override
+    public int getTotalCompactions()
     {
-        return params.isEnabled();
+        return getStrategies(false).stream()
+                                   .flatMap(list -> list.stream())
+                                   .map(CompactionStrategy::getTotalCompactions)
+                                   .reduce(0, Integer::sum);
     }
 
+    @Override
     public String getName()
     {
         return name;
     }
 
-    public List<List<AbstractCompactionStrategy>> getStrategies()
+    @Override
+    public List<List<CompactionStrategy>> getStrategies()
     {
-        return getStrategies(true);
+        return getStrategies(true).stream().collect(Collectors.toList());
```

Comment:
Why the `stream` and `collect` here? It's already a non-shared list.

Reply:
Fixed in 3c3cdb1452b6fbb402c31648e918cb00cf6a2dd7.